examples/core/rgcn/hetero_rgcn.py

```diff
@@ -0,0 +1,647 @@
+"""
+This script, `hetero_rgcn.py`, trains and tests a Relational Graph
+Convolutional Network (R-GCN) model for node classification on the
+Open Graph Benchmark (OGB) dataset "ogbn-mag". For more details on
+"ogbn-mag", please refer to the OGB website:
+(https://ogb.stanford.edu/docs/linkprop/)
+
+Paper [Modeling Relational Data with Graph Convolutional Networks]
+(https://arxiv.org/abs/1703.06103).
+
+
+This flowchart describes the main functional sequence of the provided example.
+main
+│
+├───> prepare_data
+│     │
+│     └───> Load and preprocess dataset
+│
+├───> rel_graph_embed [HIGHLIGHT]
+│     │
+│     └───> Generate graph embeddings
+│
+├───> Instantiate EntityClassify (RGCN model)
+│     │
+│     ├───> RelGraphConvLayer (input to hidden)
+│     │
+│     └───> RelGraphConvLayer (hidden to output)
+│
+└───> train
+      │
+      │
+      ├───> Training loop
+      │     │
+      │     ├───> EntityClassify.forward (RGCN model forward pass)
+      │     │
+      │     └───> test
+      │           │
+      │           ├───> MultiLayerFullNeighborSampler
+      │           │
+      │           └───> EntityClassify.evaluate
+      │
+      └───> Logger.add_result
+"""
+import argparse
+import itertools
+import sys
+
+import dgl
+import dgl.nn as dglnn
+
+import psutil
+
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl import AddReverse, Compose, ToSimple
+from dgl.nn import HeteroEmbedding
+from ogb.nodeproppred import DglNodePropPredDataset, Evaluator
+from tqdm import tqdm
+
+v_t = dgl.__version__
+
+
+def prepare_data(args, device):
+    dataset = DglNodePropPredDataset(name="ogbn-mag")
+    # Get train/valid/test index.
+    split_idx = dataset.get_idx_split()
+
+    # - graph: dgl graph object.
+    # - label: torch tensor of shape (num_nodes, num_tasks).
+    g, labels = dataset[0]
+
+    # Flatten the labels for "paper" type nodes. This step reduces the
+    # dimensionality of the labels. We need to flatten the labels because
+    # the model requires a 1-dimensional label tensor.
+    labels = labels["paper"].flatten()
+
+    # Apply transformation to the graph.
+    # - "ToSimple()" removes multi-edge between two nodes.
+    # - "AddReverse()" adds reverse edges to the graph.
+    # This could help model learn more information from the graph.
```

Comment:
nit: Is this comment always true? maybe we don't need to mention this sentence here.

Reply:
The truth of the comment is task-specific. I'll remove this comment here.